pyccel/ast/operators.py

```diff
@@ -206,7 +204,8 @@ def __init__(self, arg):
     @staticmethod
     def _calculate_dtype(*_args):
         """ Sets the dtype and precision
-        They are chosen to match the argument
+        They are chosen to match the argument unless the class has
+        a _dtype or _precision member
```

Comment:
Is this new or is it undoing corrections that you already did? I'm not sure this description is still accurate